### ISSUE TITLE
[Feature] Ability to create and use custom fields

### ISSUE THREAD

**Issue by rich-howell** [actualbudget/actual#504]
### Discussed in https://github.com/actualbudget/actual/discussions/398

<div type='discussions-op-text'>

<sup>Originally posted by **caesartcs** December 10, 2022</sup>
**Creating custom fields would be a a great feature**

As an example, keeping record of home renovation expenses - examples:

- (**Contractor**) Contractors name
- (**Location**) where I bought materials from
- (**Subtotal**) actual cost of a product I purchased (w/o tax)
- (**Total**) actual total cost of a product I purchased (w tax)

I read here that someone suggested Tags as an idea. That could help some of it. But I still think custom fields can greatly help 👍 </div>

**Comment by MatissJanis** [comment]
The notes field can be used to add this information. We are not planing to expand Actual to facilitate other types of "custom" information as it goes against the simplicity of the product and doesn't solve a problem most users will have.

But please let me know if you disagree and why the notes field would not work for this use-case.

**Comment by j-f1** [comment]
One thing that I would kind of like to have is a location field on payees. I’ve been tagging the names of restaurants and such with the location in brackets (for future reference), but it would be nice to have that as a built-in feature. Not sure if it would be common enough to be worth adding though.

**Comment by shall0pass** [comment]
> One thing that I would kind of like to have is a location field on payees. 

If this was able to help do automatic payee fill on mobile based on current location, I'd be in favor 100%.  IMO, that's a killer feature in ynab4.

**Comment by ben-davis** [comment]
Just throwing in my use for this: tagging transactions as tax deductible. I'd love to use Actual to handle my expense tracking rather than my current process of exporting everything to Google Sheets and tagging transactions there. I suppose I could just have unique categories for my deductible expenses, but then I still need an additional "Have receipt?" field.

**Comment by forresthopkinsa** [comment]
When I import from CSV, the file has a column for **Card User**, e.g. which card was actually used to make the purchase.

This is a critically important field but it gets lost when importing. There is no way to restore that data (e.g. in notes) once it has been discarded; rules naturally can't access columns that were not included during import.

tl;dr custom fields allow trivial workflows that are currently completely impossible
